Created new SONAR rule (Remove or make private all the public getter methods and properties that make Services available to Controllers)

sbommineni opened the pull request:
Please review the below new SONAR rule code and marge into master branch

Remove or make private all the public getter methods and properties that make Services available to Controllers

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/hybris/HybrisControllerHtmlUsageCheck.java:
```diff
@@ -34,7 +34,7 @@ protected boolean isTargetedType(ClassTree type) {
         return TARGET_TYPE_NAME_MATCH.matcher(type.simpleName().name()).matches();
-    }
+        }
 
```
Do we need this change as part of this pull request?

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/generic/AbstractPublicGetterUsageCheck.java:
```diff
@@ -0,0 +1,30 @@
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.*;
+
+public abstract class AbstractPublicGetterUsageCheck extends BaseTreeVisitor implements JavaFileScanner {
```
Need to understand the purpose of this Abstract class?

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/generic/GenericGetterUsageCheck.java:
```diff
@@ -0,0 +1,86 @@
+
+import java.util.regex.Pattern;
+
+@Rule(key = "GenericGetterUsageCheck", priority = Priority.MINOR, name = "Architectural layer should not be skipped", tags = { "hybris" }, description = "Use objects from immediate layer instead of skipping layers")
```
Need to update the name "Architectural layer should not be skipped". May be something which talks about public getter.

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/generic/GenericGetterUsageCheck.java:
```diff
@@ -0,0 +1,86 @@
+        for (ModifierKeywordTree modifier : tree.modifiers().modifiers()) {
+            if (modifier.modifier().equals(Modifier.PUBLIC)) {
+                if (logger.isDebugEnabled()) {
+                    logger.info("Tree modifier name :" + modifier);
```
Should this be debug logging? seem like we are checking for debug level.

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/generic/GenericGetterUsageCheck.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Override
+    protected boolean isPublicGetter(MethodTree tree) {
+        for (ModifierKeywordTree modifier : tree.modifiers().modifiers()) {
```
Does this method give false positive for non getter public method? Should we check for "get" prefix in the method? Also may be we need to check return type of the method to check if return type is a restricted type.

divyakumarjain commented on src/main/java/org/divy/sonar/check/java/generic/GenericGetterUsageCheck.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @Override
+    protected boolean isPublicGetter(MethodTree tree) {
```
Should this method be renamed to isPublicGetterForRestrictedType

divyakumarjain commented on src/main/resources/org/sonar/l10n/java/rules/squid/GenericGetterUsageCheck_java.html:
```diff
@@ -0,0 +1,20 @@
+<h2>Compliant Solution</h2>
+<pre>
+@Facade
+private String getService() { // Compliant
```
Should we document a compliant method which uses service internally and returns a non-service result?

sbommineni replied:
Done

divyakumarjain commented on src/test/files/facade/PublicServiceGetterTestSample.java:
```diff
@@ -0,0 +1,11 @@
+import org.sonar.java.checks.verifier.JavaCheckVerifier; // Compliant
+
+@Facade
+public class PublicServiceGetterTestSample {
```
There is no Annotation like @Facade

divyakumarjain commented on src/test/files/facade/PublicServiceGetterTestSample.java:
```diff
@@ -0,0 +1,11 @@
+@Facade
+public class PublicServiceGetterTestSample {
+
+    void compliantMethodService() { // Compliant
```
Please make this method public and run the test.

sbommineni commented:
Addressed all review comments a part from "
Should   we document a compliant method which uses service internally and returns a   non-service result? "
--
